templates/guest_book/gulpfilelocal.js

```diff
@@ -0,0 +1,120 @@
+const gulp = require("gulp");
+const fs = require('fs');
+const args = require("yargs").argv;
+const { Account, SimpleKeyStoreSigner, InMemoryKeyStore, KeyPair, LocalNodeConnection, NearClient, Near } = require('nearlib');
+
+gulp.task('copyfiles', function () {
+  return gulp.src('./assembly/**/*')
+    .pipe(gulp.dest('./out/'));
+});
+
+gulp.task("build:model", callback => {
+  const asc = require("assemblyscript/bin/asc");
+  asc.main([
+    "model.ts",
+    "--baseDir", "./out",
+    "--nearFile", "../out/model.near.ts",
+    "--measure"
+  ], callback);
+});
+
+gulp.task("build:bindings", ["build:model"], callback => {
+  const asc = require("assemblyscript/bin/asc");
+  asc.main([
+    "main.ts",
+    "--baseDir", "./out",
+    "--binaryFile", "../out/main.wasm",
+    "--nearFile", "../out/main.near.ts",
+    "--measure"
+  ], callback);
+});
+
+gulp.task("build:all", ["build:bindings"], callback => {
+  const asc = require("assemblyscript/bin/asc");
+  asc.main([
+    "../out/main.near.ts",
+    "--baseDir", "./out",
+    "-O3",
+    "--binaryFile", "../out/main.wasm",
+    "--sourceMap",
+    "--measure"
+  ], callback);
+});
+
+gulp.task('build', function(done) {
+  const runSequence = require('run-sequence');
+  runSequence('copyfiles', 'build:all', function() {
+    done();
+  });
+});
+
+gulp.task('deploy', function(done) {
+  // Find keys/ account id
+  const keyPath = './neardev/devkey.json';
+  if (!fs.existsSync(keyPath)) {
+    console.error("Deployment failed. File ./neardev/devkey.json does not exist. Please add this file and try again. Please make sure file contains public_key, secret_key, and account_id.")
+    return;
+  }
+  const rawKey = JSON.parse(fs.readFileSync(keyPath));
+  if (!rawKey.public_key || !rawKey.secret_key || !rawKey.account_id) {
+    console.error("Deployment failed. neardev/devkey.json format problem. Please make sure file contains public_key, secret_key, and account_id")
+    return;
+  }
+  const keyPair = new KeyPair(rawKey.public_key, rawKey.secret_key);
+  const accountId = rawKey.account_id;
+  const keyStore = new InMemoryKeyStore();
+  keyStore.setKey(accountId, keyPair);
+
+  // Determine which url to use
+  const nodeUrl = getNodeUrl(args);
+  const localNodeConnection = new LocalNodeConnection(nodeUrl);
+  const nearClient = new NearClient(new SimpleKeyStoreSigner(keyStore), localNodeConnection);
+  const near = new Near(nearClient);
```

Comment:
this is way too much boilerplate for relatively small task. It's not blocking this PR, but I think generally all this stuff should be as simple as:

```
const near = new Near({ nodeUrl, keyPath: 'neardev/devkey.json'  });
```

or like:

```
const near = nearlib.connect({ nodeUrl, keyPath: 'neardev/devkey.json'  });
```

Reply:
I don't think using a keypath in this particular api is a good option, since we have multiple methods of storing keys and also multiple potential signers.

Reply:
That's a convenience option. If anyone needs some custom signer they can go and create the whole stack as it's created now. I'm not suggesting removing this possibility. But basic usage should be easy and not require tons of boilerplate. Ideally it should look something like this:

```
const near = await nearlib.connect(); // Connects to local node (if it is running) or default gateway if node isn't running with default key storage that makes sense for the context (our wallet for browser, 'neardev/devkey.json` for Node, etc).
const contract = await nearlib.loadContract('thebestcontract.near');
await contract.doSomething();
```

I think it's a good idea for APIs to follow Alan Kay's “Simple things should be simple, complex things should be possible.” principle.